base/multidimensional.jl

```diff
@@ -358,8 +358,12 @@ module IteratorsMD
         CartesianIndex()
     end
     @propagate_inbounds function Base.getindex(iter::CartesianIndices{N,R}, I::Vararg{Int, N}) where {N,R}
-        CartesianIndex(getindex.(iter.indices, I))
+        CartesianIndex(_get_cartesianindex.(iter.indices, I))
     end
+    # specialize to gives better hint to compiler to enable SIMD (#42115)
+    @inline _get_cartesianindex(::OneTo, i::Int) = i
+    @inline _get_cartesianindex(::Base.IdentityUnitRange, i::Int) = i
+    @inline _get_cartesianindex(r, i) = getindex(r, i)
```

Comment:
I think it would make sense to also have `@inline _get_cartesianindex(::Base.Slice, i::Int) = i`, since this is what slice does when inbounds.